### PR TITLE
Perf/optimize load time

### PR DESCRIPTION
**Pull request by paphonb**
## Why did you create this PR
Our bundle has grown a lot throughout the year, resulting in slow load time. The current production version has over 318kB of critical JS (not counting other resources) and scores horribly on [lighthouse](https://googlechrome.github.io/lighthouse/viewer/?gist=799226755b6e8a5d43213369503e9cc6).

## What did you do
- Dropped lodash
- Lazy load a bunch of stuffs in schedule page

There are still a lot left to do. If you have any idea, please comment or push to this PR.

### CODE REVIEW

**Comment by new5558** [comment]
I thought lodash now has tree shaking and should not impact performance anymore.

**Comment by paphonb** [comment]
> I thought lodash now has tree shaking and should not impact performance anymore.

Apparently not. Just removing it reduces the bundle size by 24kB.

**Comment by new5558** [comment]
> > I thought lodash now has tree shaking and should not impact performance anymore.
> 
> Apparently not. Just removing it reduces the bundle size by 24kB.

Then, please write test for removed functions

**Comment by paphonb** [comment]
> > > I thought lodash now has tree shaking and should not impact performance anymore.
> > 
> > 
> > Apparently not. Just removing it reduces the bundle size by 24kB.
> 
> Then, please write test for removed functions

Done

File: src/modules/Schedule/components/ScheduleTable/index.tsx [hunk 1]
```diff
@@ -1,51 +1,43 @@
...
 
 export interface ScheduleTableProps {
   courseCart: CourseCart
   overlappingCourses: CourseOverlapMap
 }
 
-const Layout = styled.div`
-  padding-bottom: 100px;
-`
+const InteractiveScheduleTable = lazy(() => import('./InteractiveScheduleTable'))
```
Comment (saenyakorn) [inline src/modules/Schedule/components/ScheduleTable/index.tsx:16]: why dont you use `dynamic` na?

Comment (saenyakorn) [reply src/modules/Schedule/components/ScheduleTable/index.tsx:16]: How about https://nextjs.org/docs/advanced-features/dynamic-import#with-suspense ?

Comment (paphonb) [reply src/modules/Schedule/components/ScheduleTable/index.tsx:16]: The suspense prop doesn't exist in our version

Comment (paphonb) [reply src/modules/Schedule/components/ScheduleTable/index.tsx:16]: It's probably added in next 12 , which you haven't merge yet

File: src/common/components/RichTextEditor/index.tsx [hunk 2]
```diff
@@ -1,39 +1,75 @@
...
+
+export const RichTextEditor = forwardRef<RichTextEditorRef | undefined, RichTextEditorProps>(
+  ({ id, defaultValue = INITIAL_CONTENT, onChange }, ref) => {
+    const theme = useTheme()
+    const { resetEditor, setValue: setEditorValue } = usePlateActions(id)
+
+    const setValue = useCallback(
+      (nodes: TNode[]) => {
+        setEditorValue(nodes, id)
+        resetEditor(id)
```
Comment (Anon-136) [inline src/common/components/RichTextEditor/index.tsx:32]: yo, what does `resetEditor(id)` do?
@saenyakorn i'm just curious

**Review by Anon-136 — APPROVED** [review]
Looking god to me as always!

File: src/modules/Schedule/components/ColorPicker/utils/getNewColor.ts [hunk 3]
```diff
@@ -28,9 +29,9 @@ export function getNewColor(currentItems: CourseCartItem[], newCourse: Course) {
       return 'secondary'
   }
   const currentColors = currentItems.map((item) => item.color)
-  const newColor = difference(OTHER_COLORS, currentColors)[0]
+  const newColor = difference(OTHER_COLORS as readonly ScheduleColor[], currentColors)[0]
```
[the remark this answers is not in the document]
Comment (paphonb) [reply src/modules/Schedule/components/ColorPicker/utils/getNewColor.ts:32]: Yes we do. It fails without readonly

File: src/utils/stubFalse/test.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import assert from 'assert'
+
+import { stubFalse } from '@/utils/stubFalse'
+
+describe('sum', function () {
+  it('should return false', function () {
```
[the remark this answers is not in the document]
Comment (paphonb) [reply src/utils/stubFalse/test.ts:6]: These tests are copied from loads and I don't want to change much

File: src/common/components/RichTextEditor/index.tsx (hunk 2 above)
Comment (saenyakorn) [reply src/common/components/RichTextEditor/index.tsx:32]: Just `setEditorValue` can't display the change on the web (Dont know why). You need to reset the editor.

Comment (Anon-136) [reply src/common/components/RichTextEditor/index.tsx:32]: ok, magic

**Pull request merged**